src/Optimiser.cpp

```diff
@@ -99,6 +99,16 @@ double Optimiser::optimise_partition(vector<MutableVertexPartition*> partitions,
     if (graphs[layer]->vcount() != n)
       throw Exception("Number of nodes are not equal for all graphs.");
 
+  // Get the map of original communities for fixed nodes
+  vector< map<size_t, size_t> > original_fixed_memberships(nb_layers);
```

Comment:
I don't think we need this for all layers. Effectively, all layers have identical membership information, so we can stick to only a single `map<size_t, size_t>`.

Reply:
done